WordPress/Classes/ViewRelated/Aztec/ViewControllers/AztecPostViewController.swift

```diff
@@ -1018,32 +1023,37 @@ extension AztecPostViewController {
         postEditorStateContext.updated(isBeingPublished: true)
 
         // Finally, publish the post.
-        publishPost() { uploadedPost, error in
-            self.postEditorStateContext.updated(isBeingPublished: false)
-            SVProgressHUD.dismiss()
-
-            let generator = UINotificationFeedbackGenerator()
-            generator.prepare()
-
-            if let error = error {
-                DDLogError("Error publishing post: \(error.localizedDescription)")
-
-                SVProgressHUD.showDismissibleError(withStatus: self.postEditorStateContext.publishErrorText)
-                generator.notificationOccurred(.error)
-            } else if let uploadedPost = uploadedPost {
-                self.post = uploadedPost
-
-                generator.notificationOccurred(.success)
-            }
-
-            if dismissWhenDone {
-                self.dismissOrPopView(didSave: true)
-            } else {
-                self.createRevisionOfPost()
-            }
+        publishPost { [weak self] uploadedPost, error in
+            self?.shouldAutoSave = false
+            self?.publishPostCompletion(dismissWhenDone: dismissWhenDone, post: uploadedPost, error)
+        }
+    }
+    
+    fileprivate func publishPostCompletion(dismissWhenDone: Bool, post: AbstractPost?, _ error: Error?) {
```

Comment:
I'm not certain about refactoring this into a new method. I think it's the right idea, as this closure is too long, but I don't think breaking out everything but the `self?.shouldAutoSave = false` accomplishes enough. We could probably break it into several methods, such as one to deal with the generator, while keeping `dismissWhenDone`, `post`, and `error` inline. That may be out of the scope of this PR.

I dunno; I don't exactly feel strongly about this, I just think the right solution to this is a bit more complex. Let me know if you disagree 😀